xarray/namedarray/core.py

```diff
@@ -338,7 +391,12 @@ def sizes(self) -> dict[Hashable, int]:
         """Ordered mapping from dimension names to lengths."""
         return dict(zip(self.dims, self.shape))
 
-    def _replace(self, dims=_default, data=_default, attrs=_default) -> Self:
+    def _replace(
+        self,
+        dims: DimsInput | Default = _default,
+        data: T_DuckArray | np.typing.ArrayLike | Default = _default,
+        attrs: AttrsInput | Default = _default,
+    ) -> Self:
```

Comment:
If the duck array type differs from Self the generic part of the return type should change as well.

Probably need a second TypeVar for this.